ctapipe/image/muon/tests/test_muon_ring_finder.py

```diff
@@ -1,113 +1,42 @@
-    # ... add Status Flag?
-    assert output.ring_radius is not np.NaN
-    assert output.ring_phi is not np.NaN
-    assert output.ring_inclination is not np.NaN
-
-
-def test_ChaudhuriKunduRingFitter():
-
-    geom = CameraGeometry.from_name('LSTCam')
-    focal_length = u.Quantity(28, u.m)
-
-    ring_radius = u.Quantity(0.4, u.m)  # make sure this is in camera coordinates
-    ring_width = u.Quantity(0.03, u.m)
-    center_x = u.Quantity(-0.2, u.m)
-    center_y = u.Quantity(-0.3, u.m)
-
-    muon_model = RingGaussian(
-        x=center_x, y=center_y,
-        sigma=ring_width, radius=ring_radius,
+from ctapipe.image.muon import MuonRingFitter
+from ctapipe.image import tailcuts_clean, toymodel
+
+def test_chaudhuri_kundu_fitter():
+    center_xs = 0.3 * u.m
+    center_ys = 0.6 * u.m
+    ring_radius = 0.3 * u.m
+    ring_width = 0.05 * u.m
+    muon_model = toymodel.RingGaussian(
+        x=center_xs,
+        y=center_ys,
+        radius=ring_radius,
+        sigma=ring_width,
     )
 
+    geom = CameraGeometry.from_name("FlashCam")
+    flashcam_focal_length = u.Quantity(16, u.m)
     image, _, _ = muon_model.generate_image(
         geom, intensity=1000, nsb_level_pe=5,
     )
+    mask = tailcuts_clean(geom, image, 10, 12)
+    x = (geom.pix_x / flashcam_focal_length) * u.rad
+    y = (geom.pix_y / flashcam_focal_length) * u.rad
+    img = image * mask
 
-    clean_mask = tailcuts_clean(
-        geom, image, boundary_thresh=5, picture_thresh=10
-    )
+    #call specific method with fit_method, teldes needed for Taubin fit
+    muonfit = MuonRingFitter(teldes = None, fit_method="chaudhuri_kundu")
+    muon_ring_parameters = muonfit.fit(x, y, img)
+    xc_fit = muon_ring_parameters.ring_center_x
+    yc_fit = muon_ring_parameters.ring_center_y
+    r_fit = muon_ring_parameters.ring_radius
 
-    fitter = muon_ring_finder.ChaudhuriKunduRingFitter()
-
-    x = geom.pix_x / focal_length * u.rad
-    y = geom.pix_y / focal_length * u.rad
-
-    # fit 3 times, first iteration use cleaning, after that use
-    # distance to previous fit result
-    result = None
-    for _ in range(3):
-        if result is None:
-            mask = clean_mask
-        else:
-            dist = np.sqrt((x - result.ring_center_x)**2 + (y - result.ring_center_y)**2)
-            ring_dist = np.abs(dist - result.ring_radius)
-            mask = ring_dist < (result.ring_radius * 0.4)
-
-        result = fitter.fit(x[mask], y[mask], image[mask])
-
-    assert np.isclose(
-        result.ring_radius.to_value(u.rad), ring_radius / focal_length,
-        rtol=0.05,
-    )
-    assert np.isclose(
-        result.ring_center_x.to_value(u.rad), center_x / focal_length,
-        rtol=0.05
-    )
-    assert np.isclose(
-        result.ring_center_y.to_value(u.rad), center_y / focal_length,
-        rtol=0.05,
-    )
+    assert np.isclose(xc_fit * flashcam_focal_length / u.m / u.rad, center_xs / u.m, 1e-1)
```

Comment:
you can use astropys unit aware version of isclose: `u.isclose`, also, like before: use `.to_value` to convert quantities to plain numbers.

Reply:
Currently all results (e.g. the hillas parameters) are in camera coordinates, this looks like the xc_fit is in CameraCoordiantes but expressed in degrees using a custom conversion. This should not be the case.